src/main/ArgsParser.java

```diff
@@ -0,0 +1,80 @@
+package main;
+
+import main.algorithms.Algorithm;
+import main.algorithms.ShiftAlgorithmImpl;
+import main.algorithms.UnicodeAlgorithmImpl;
+
+public class ArgsParser {
+
+    private String mode;
+    private String data;
+    private int key;
+    private String in;
+    private String out;
+    private Algorithm algorithm;
+
+    public ArgsParser() {
+        mode = "enc";
+        data = "";
+        key = 0;
+        out = "";
+    }
+
+    public void parseArgs(String[] args) {
+        for (int i = 0; i < args.length; i += 2) {
+            switch (args[i]) {
+                case "-mode" -> mode = args[i + 1];
+                case "-key" -> {
+                    try {
```

Comment:
Can you move this try-catch block to another method?

Reply:
done.